Assets/ConsentManagementProvider/Plugins/iOS/Source/SwiftBridge.swift

```diff
@@ -0,0 +1,334 @@
+
+// MARK: - Bridge config
+    @objc public func addTargetingParam(campaignType: Int, key: String, value: String){
+        switch CAMPAIGN_TYPE(rawValue: campaignType) {
+        case .GDPR: gdprTargetingParams[key]=value
+
+        case .IOS14: break
+
+        case .CCPA: ccpaTargetingParams[key]=value
+
+        case .none:
+            print("Incorrect campaignType on addTargetingParam")
+        }
+    }
+    
+    @objc public func configLib(
+        accountId: Int,
+        propertyId: Int,
+        propertyName: String,
+        gdpr: Bool,
+        ccpa: Bool,
+        language: SPMessageLanguage,
+        gdprPmId: String,
+        ccpaPmId: String) {
+            self.config = { Config(
+                accountId: accountId,
+                propertyId: propertyId,
+                propertyName: propertyName,
+                gdpr: gdpr,
+                ccpa: ccpa,
+                att: true,
+                language: language,
+                gdprPmId: gdprPmId,
+                ccpaPmId: ccpaPmId
+            )}()
+        }
+    
+    @objc public func addCustomVendor(vendor: String) {
+        config.vendors.append(vendor)
+    }
+
+    @objc public func addCustomCategory(category: String) {
+        config.categories.append(category)
+    }
+
+    @objc public func addCustomLegIntCategory(legIntCategory: String) {
+        config.legIntCategories.append(legIntCategory)
+    }
+    
+    @objc public func clearCustomArrays() {
+        config.vendors = []
+        config.categories = []
+        config.legIntCategories = []
+    }
+    
+    @objc public func dispose() {
+        callbackDefault = nil
+        callbackOnConsentReady = nil
+        callbackOnConsentUIReady = nil
+        callbackOnConsentAction = nil
+        callbackOnConsentUIFinished = nil
+        callbackOnErrorCallback = nil
+        callbackOnSPFinished = nil
+        callbackOnSPUIFinished = nil
+        callbackOnCustomConsent = nil
+    }
+ 
+// MARK: - Manage lib
+    @objc public func loadMessage(authId: String? = nil) {
+        print("PURE SWIFT loadMessage")
+        consentManager.loadMessage(forAuthId: authId)
+    }
+
+    @objc public func onClearConsentTap() {
+        SPConsentManager.clearAllData()
+        myVendorAccepted = .Unknown
+    }
+    
+    @objc public func onGDPRPrivacyManagerTap() {
+        consentManager.loadGDPRPrivacyManager(withId: config.gdprPmId!)
+    }
+    
+    @objc public func onCCPAPrivacyManagerTap() {
+        consentManager.loadCCPAPrivacyManager(withId: config.ccpaPmId!)
+    }
+
+
+    @objc public func customConsentToGDPR() {
+        consentManager.customConsentGDPR(
+            vendors: config.vendors,
+            categories: config.categories,
+            legIntCategories: config.legIntCategories){contents in
+                self.print(contents)
+                self.runCallback(callback: self.callbackOnCustomConsent, arg: contents.toJSON())
+            }
```

Comment:
@Nevazhnovu I'm curious to know why you chose adding `vendors`, `categories` and `legIntCategories` to Config, instead of passing it as parameters to `customConsentToGDPR` like the SDK does.

Reply:
@andresilveirah we do it because we add them one by one to later bypass them altogether to Unity on demand. 
I need to cache them here in the bridge.